packages/babel-generator/test/fixtures/flow/arrow-functions/output.js

```diff
@@ -10,4 +10,6 @@ const bar5 = (x): string => {};
 
 const bar6 = (x: number) => {};
 
-const bar7 = <T>(x) => {};
\ No newline at end of file
+const bar7 = <T>(x) => {};
+
+const bar8 = <T>(x: T): ((T) => T[]) => y => [x, y];
```

Comment:
hmm, I see. Maybe the correct output should be the one in the code comment.

Reply:
I was look into dropping the parens around `(T)` later, initial experiment looked like it touched a bunch of other fixtures.